File: VSCode_Java/leetcode/editor/en/1.two-sum.java
package leetcode.editor.en;/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mem = new HashMap<>();
        for(int i=0; i < nums.length; i++){
            if(mem.containsKey(nums[i])){
                int left = mem.get(nums[i]);
                return new int[]{left, i};
            }
            else
                mem.put(target-nums[i], i);
        }
        return new int[2];
    }
}
// @lc code=end

File: 22FALL/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums) - 1):
            if target -  nums[i] in nums[i+1:len(nums)]:
                return [i, nums[i+1:len(nums)].index(target - nums[i])]

# @lc code=end

